clients/src/main/java/org/apache/kafka/clients/NetworkClientUtils.java

```diff
@@ -56,20 +56,19 @@ public static boolean awaitReady(KafkaClient client, Node node, Time time, long
         long startTime = time.milliseconds();
         long expiryTime = startTime + timeoutMs;
 
-        if (!isReady(client, node, startTime))
-            client.ready(node, startTime);
+        if (isReady(client, node, startTime) ||  client.ready(node, startTime))
```

Comment:
Not sure what the benefit is here. If `isReady` is false, `ready` will also return false after initiating a connection. So both disjuncts will be true or false together, and hence the disjunction is not adding any value.